test/functional/groups/invites_controller_test.rb

```diff
@@ -65,4 +65,67 @@ def test_create_email_invite
     assert req.valid?
   end
 
+  def test_request_to_join_us_notifies_recipient
+    login_as @user
+    recipient = FactoryGirl.create(:user)
+
+    assert_difference 'RequestNotice.count' do
+      post :create, group_id: @group.to_param, recipients: recipient.name
+    end
+
+    notice = RequestNotice.last
+    assert_equal recipient.id, notice.user_id
+    assert_equal 'request_to_join_us', notice.data[:title]
+  end
+
+  def test_request_to_join_us_notifies_all_recipients
+    login_as @user
+    recipient1 = FactoryGirl.create(:user)
+    recipient2 = FactoryGirl.create(:user)
+
+    assert_difference('RequestNotice.count', 2) do
+      post :create, group_id: @group.to_param,
+        recipients: "#{recipient1.name}, #{recipient2.name}"
+    end
+
+    notices = RequestNotice.last(2)
+    assert_equal recipient1.id, notices.first.user_id
+    assert_equal recipient2.id, notices.last.user_id
+    assert_equal 'request_to_join_us', notices.first.data[:title]
+    assert_equal 'request_to_join_us', notices.last.data[:title]
+  end
+
+  def test_request_to_join_us_notifies_only_valid_recipients
+    login_as @user
+    recipient = FactoryGirl.create(:user)
+
+    assert_difference 'RequestNotice.count' do
+      post :create, group_id: @group.to_param,
+        recipients: "#{recipient.name}, #{@user.name}"
+    end
+
+    notice = RequestNotice.last
+    assert_equal recipient.id, notice.user_id
+    assert_equal 'request_to_join_us', notice.data[:title]
+  end
+
+  def test_invite_group_to_network_notifies_group_member
```

Comment:
Actually yes, if the group has no council, all members should be notified. Quite a lot of notifications for large group, but on the other hand we have not another possibility to invite a group rather than that.

How people from the closed group would know about the invite? (When you can not contact anybody personally to ask for council creation)

What about tests - not sure that I got the point, will double check it little bit later :)